core/src/main/scala/org/apache/spark/scheduler/TaskSchedulerImpl.scala

```diff
@@ -388,14 +388,16 @@ private[spark] class TaskSchedulerImpl(
       val execId = shuffledOffers(i).executorId
       val host = shuffledOffers(i).host
       val taskSetRpID = taskSet.taskSet.resourceProfileId
+      val prof = sc.resourceProfileManager.resourceProfileFromId(taskSetRpID)
+      val targetExecutorRpID = prof.targetExecutorRpId
```

Comment:
I think once we do https://github.com/apache/spark/pull/37268/files#r934220131, we'd no longer need this.